src/ch/ch_hotplug.c

```diff
@@ -0,0 +1,194 @@
+/*
+ * ch_hostdev.h: Cloud Hypervisor hotplug support
+ *
+ * Copyright Intel Corp. 2020
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library.  If not, see
+ * <http://www.gnu.org/licenses/>.
+ */
+
+#include "ch_cgroup.h"
+#include "ch_domain.h"
+#include "ch_hotplug.h"
+#include "ch_monitor.h"
+#include "ch_process.h"
+#include "domain_audit.h"
+
+#define VIR_FROM_THIS VIR_FROM_CH
+
+/**
+ * virCHDomainSelectHotplugVcpuEntities:
+ *
+ * @def: domain definition
+ * @nvcpus: target vcpu count
+ *
+ * Validate vcpu entities will be able to be enabled/disabled.
+ *
+ * Returns the true if the nvcpus can be set, false if not.
+ */
+static bool
+virCHDomainSelectHotplugVcpuEntities(virDomainDefPtr def,
```

Comment:
I think this function validates if the nvcpus is with in allowable ranges. Maybe `virCHDomainValidateHotplugVcpuEntities` be a better name?

Reply:
Yea, I think I was trying to hard to keep the qemu naming convention for similar functions. Fixing.